ksqldb-common/src/main/java/io/confluent/ksql/serde/FormatInfo.java

```diff
@@ -33,6 +34,8 @@ public final class FormatInfo {
   private final String format;
   private final ImmutableMap<String, String> properties;
 
+  private static final String SCHEMA_ID = "schemaId";
```

Comment:
Ah I see, this constant is currently defined in the ksqldb-serde package but you need it from ksqldb-common here. I think it's best to avoid duplication, in case one evolves without the other in the future (unlikely but possible). What do you think about changing `copyWithoutSchemaId()` to take the string key to filter out as an argument? For example, `copyWithoutProperty(String propName)`

Reply:
Adding dependency introduces a circular dependency between `ksqldb-serde` and `ksqldb-common` modules.

`copyWithoutProperty(String propName)` will resolve that dependency issue since the constant will be passed from the `InternalFormats` class from ksqldb-engine